baseline.yaml

```diff
@@ -649,6 +649,33 @@ criteria:
     security_insights_value: # TODO
     scorecard_probe: # TODO
 
+      - id: OSPS-DO-18   
+    maturity_level: 2
+    category: Documentation
+    criteria: |
+       The project MUST perform threat modeling and
+       attack surface analysis to understand and protect
+       against critical code paths, functions, and interactions 
+       with the system.
+    objective: |
+       Projects need to conduct threat modeling, attack 
+       surface analysis, and data-flow analysis in order
+       to understand, document, and plan protections
+       to avoid future explotation of threats and weaknesses.
+
+       Identifying these areas helps the project plan on
+        reducing potential attack surface and to harden 
+        the software from specific attacks.
+    implementation: |
+       Create a status check that checks the project's
+       version control system for documented threat 
+       modeling, attack surface analysis, and data flow analysis.
```

Comment:
The status check is the job tools that want to automate the compliance check, the implementation should guide the project to comply with the criteria, I think should be summarized as:

1. Write the threat model
2. Publish it somewhere
3. Leave behind some pointers linking the repository to it, form example in security insights or using an [SPDX external reference](https://spdx.github.io/spdx-spec/v3.0.1/model/Core/Vocabularies/ExternalRefType/) of type `securityThreatModel`.

Reply:
```suggestion
       Create a status check that checks the project's
       version control system for documented threat 
       modeling, attack surface analysis, and data flow analysis.
       
       The location of the written threat model MAY be expressed using [the `security-artifacts.threat-model`
       fields in `SECURITY-INSIGHTS.yaml`](https://github.com/ossf/security-insights-spec/blob/main/specification/security-artifacts.md),
       or via a [SPDX external reference](https://spdx.github.io/spdx-spec/v3.0.1/model/Core/Vocabularies/ExternalRefType/)
       of type `securityThreatModel`, for example.
```

Reply:
The `artifacts.threat-model` field has 3 sub-fields:

* `threat-model-created` (bool)
* `evidence-url` (array of IRI / RFC 3987)
* `comment` (<560 chars)

I'd expect many `evidence-url`s to be outside of VCS, but the idea is to put the *pointer* into VCS, so we can find and fetch the external resource.